scripts/gen_universe.py

```diff
@@ -705,6 +709,39 @@ def validate_repo_with_schema(repo_json_data, repo_version):
     return errors
 
 
+def _populate_dcos_version_json_to_folder(dcos_version, outdir):
+    """Populate the repo-up-to-<dcos-version>.json file to a folder.
+    The folder structure would be :
+        <dcos-version>
+            -package
+                -<name-of-package1>.json
+                -<name-of-package2>.json
+
+    :param dcos_version: The version of DC/OS file to process.
+    :type dcos_version: str
+    :param outdir: Path to the directory to use to store all universe objects
+    :type outdir: str
+    :return: None
+    """
+    repo_dir = outdir / dcos_version / 'package'
+    pathlib.Path(repo_dir).mkdir(parents=True)
+    repo_file = pathlib.Path(outdir / 'repo-up-to-{}.json'.format(dcos_version))
+    with repo_file.open('r',  encoding='utf-8') as f:
+        data = json.loads(f.read())
+        packages_dict = {}
+
+        for package in data.get('packages'):
+            package_name = package.get('name')
+            package_list = packages_dict.get(package_name, [])
+            package_list.append(package)
+            packages_dict[package_name] = package_list
+
+        for package_name, package_list in packages_dict.items():
```

Comment:
Is there a reason for this separation? Seems like you could just do everything in a single for loop, no?

Reply:
Yes and No.
I don't think i can do in single for loop. In the first for loop I am iterating over each package and gathering the definitions. This has to be completely done before I write each package (with all its gathered definitions) to a single json file.
However, as all the packets are currently organized in a sequential fashion, we can rely on that to parse all the packages and dump them to json whenever a new package is encountered. But i think its more safe to rely on that.

Reply:
Fair enough